### PR TITLE
chore(route-monitor-operator): add console monitor [OSD-5499]

### PR DESCRIPTION
**Pull request by georgettica**
I have 2 questions for this PR:
- who should be the owners
- what should the `.spec.targetAvailabilityPercent` be?

### CODE REVIEW

**Comment by RiRa12621** [comment]
It doesn't need a separate owner.
You can't merge without the gods' approval to prod anyway
Availability seems fine, needs revisit after a while to see if we can raise it

**Comment by georgettica** [comment]
/hold
I need to add a SOP for it

File: deploy/osd-route-monitor-operator/100-rmo.RouteMonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: monitoring.openshift.io/v1alpha1
+kind: RouteMonitor
+metadata:
+  name: routemonitor-sample
```
Comment (RiRa12621) [inline deploy/osd-route-monitor-operator/100-rmo.RouteMonitor.yaml:4]:
```suggestion
name: routemonitor-sample
```

**Comment by cblecker** [comment]
> It doesn't need a separate owner.

There should be an owners file with a `reviewers` section detailing who should be point of contact for reviews. You can look at the other folders under `deploy/` for examples

> what should the `.spec.targetAvailabilityPercent` be?

Do we have a process for selecting these? Will alerts be raised based off this, or just a metric?

cc @dofinn

**Comment by RiRa12621** [comment]
Who needs that if it can't go in without "approved" anyway?

Yes alerts is what route monitor operator does

**Comment by cblecker** [comment]
> Who needs that if it can't go in without "approved" anyway?

Because change to MCC should be reviewed by subject matter experts in addition to leads.

**Comment by RiRa12621** [comment]
So you need me to tell you it's ok, so you can approve? Not very efficient imho
Let SMEs approve if they're (rightfully) supposed to evaluate and review changes.

File: deploy/osd-route-monitor-operator/100-rmo.RouteMonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: monitoring.openshift.io/v1alpha1
+kind: RouteMonitor
+metadata:
+  name: console
+spec:
+  route:
+    name: console
+    namespace: openshift-console
+  slo:
+    targetAvailabilityPercent: "99.5"
```
Comment (dofinn) [inline deploy/osd-route-monitor-operator/100-rmo.RouteMonitor.yaml:10]: This needs to be a string?

**Comment by dofinn** [comment]
We have nothing in stone that helps us define initial values for SLOs. 
Lets make sure we have some dedicated actions to follow this up to analyze if we can increase this SLO based on current data.

File: deploy/osd-route-monitor-operator/100-rmo.RouteMonitor.yaml (hunk 2 above)
Comment (georgettica) [reply deploy/osd-route-monitor-operator/100-rmo.RouteMonitor.yaml:10]: There are no floating points support in the apiserver :(

Comment (dofinn) [reply deploy/osd-route-monitor-operator/100-rmo.RouteMonitor.yaml:10]: The types defined in the CRD? 
https://github.com/openshift/route-monitor-operator/blob/master/config/crd/bases/monitoring.openshift.io_routemonitors.yaml#L55  

We could update this to be a `number` as per: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#type

Comment (georgettica) [reply deploy/osd-route-monitor-operator/100-rmo.RouteMonitor.yaml:10]: thanks @dofinn ! I'll see how we do that in json stuffz

Comment (georgettica) [reply deploy/osd-route-monitor-operator/100-rmo.RouteMonitor.yaml:10]: @dofinn seems this is not 100% supported by operator-sdk / kubebuilder as of now
see https://github.com/kubernetes-sigs/controller-tools/issues/245 as people work on shying away from floats

also in an internal discussion I tried to understand how to implement doubles in the api resource, but didn't revieve any ❤️  on it (aside of @dastergon)

**Comment by georgettica** [comment]
/unhold
changes that @NautiluX requested are in and SOP is also in

@RiRa12621 can you give a final /lgtm?
not sure if @dofinn has additional notes on this PR aside of what was raised

File: deploy/osd-route-monitor-operator/100-rmo.RouteMonitor.yaml (hunk 2 above)
Comment (dofinn) [reply deploy/osd-route-monitor-operator/100-rmo.RouteMonitor.yaml]: ooo thats interesting. nice find ! @georgettica

**Comment by georgettica** [comment]
as @NautiluX suggeste, changed NS and need a lgtm becuase of it

**Comment by georgettica** [comment]
tested on stage (I can replace my routemonitor with this one, but outcome should be similar)

**Comment by georgettica** [comment]
changing to the current CR still works and is scraped, only future concern is the message naming, which should be solved in the RMO code itself

(the value of the breach is 1 and not a good indication to how long the alert is firing)

File: deploy/osd-route-monitor-operator/100-openshift-route-monitor-operator.console.RouteMonitor.yaml [hunk 3]
```diff
@@ -8,4 +8,4 @@ spec:
     name: console
     namespace: openshift-console
   slo:
-    targetAvailabilityPercent: "99.5"
+    targetAvailabilityPercent: "99.95"
```
Comment (NautiluX) [inline deploy/osd-route-monitor-operator/100-openshift-route-monitor-operator.console.RouteMonitor.yaml:11]: Why now 99.95? According to the [SLO doc](https://docs.google.com/document/d/1MIh_4GNTFyMaGnPBrq_cxKIrWoKw7CMShq47qp5Hj48/edit#) it should be 99.5? Probably I miss some discussion here...

Comment (NautiluX) [reply deploy/osd-route-monitor-operator/100-openshift-route-monitor-operator.console.RouteMonitor.yaml:11]: turns out this was confused with the API server SLA. We'll change it back to 99.5.

**Comment by wgordon17** [comment]
Will this automatically merge once #771 is resolved?

**Comment by georgettica** [comment]
@wgordon17 If I get a +1 for this then yes

**Comment by wgordon17** [comment]
@georgettica Consider this a ➕ 1 from PM if it helps :) Not sure if you need plus one's from others ¯\_(ツ)_/¯

**Comment by cblecker** [comment]
@georgettica LGTM, but as previously commented, needs an `OWNERS` file that looks something like:
```
reviewers:
- georgettica
- RiRa12621
```

**Pull request merged**